src/commands/tail-cloudwatch-logs.js

```diff
@@ -0,0 +1,163 @@
+const _ = require("lodash");
+const { getAWSSDK } = require("../lib/aws");
+const { Command, flags } = require("@oclif/command");
+const { checkVersion } = require("../lib/version-check");
+const Promise = require("bluebird");
+const moment = require("moment");
+const inquirer = require("inquirer");
+
+class TailCloudwatchLogsCommand extends Command {
+	async run() {
+		const { flags } = this.parse(TailCloudwatchLogsCommand);
+		const { namePrefix, region, profile, filterPattern, interval } = flags;
+
+		global.region = region;
+		global.profile = profile;
+		global.filterPattern = filterPattern;
+		global.interval = interval || 1000;
+
+		checkVersion();
+
+		const AWS = getAWSSDK();
+		this.Logs = new AWS.CloudWatchLogs();
+
+		this.log(
+			`looking for CloudWatch log groups with prefix [${namePrefix}] in [${region}]`
+		);
+		const logGroupName = await this.getLogGroupName(namePrefix);
+
+		this.log(`looking for log streams for [${logGroupName}]...`);
+		const logStreamNames = await this.getLogStreamNames(logGroupName);
+
+		this.log(`polling CLoudWatch log group [${logGroupName}]...`);
+		this.log("press <any key> to stop");
+		await this.pollLogGroup(logGroupName, logStreamNames);
+
+		this.exit(0);
+	}
+
+	async getLogGroupName(namePrefix) {
+		const resp = await this.Logs.describeLogGroups({
+			logGroupNamePrefix: namePrefix
+		}).promise();
+
+		if (resp.logGroups.length === 1) {
+			this.log("only 1 match found");
+			return resp.logGroups[0].logGroupName;
+		} else if (_.isEmpty(resp.logGroups)) {
+			this.log(
+				"no matching log groups, please double check the prefix and region and try again"
+			);
+			this.exit(1);
+		} else if (resp.nextToken) {
+			this.log(
+				"too many log groups with matching prefix, please provide the full log group name and try again"
```

Comment:
Worth mentioning that the command is willing to pull up to 50 log groups

Reply:
Otherwise, the users won't know how much to trim

Reply:
good point!

Reply:
done